include/mqtt/v5_message.hpp

```diff
@@ -0,0 +1,2423 @@
+              std::accumulate(
+                  props.begin(),
+                  props.end(),
+                  0U,
+                  [](std::size_t total, property_variant const& pv) {
+                      return total + v5::size(pv);
+                  }
+              )
+          ),
+          props_(std::move(props)),
+          num_of_const_buffer_sequence_(
+              1 +                   // fixed header
+              1 +                   // remaining length
+              1 +                   // packet id
+              (
+                  reason_code ?
+                  (
+                      1 +                   // reason code
+                      1 +                   // property length
+                      std::accumulate(
+                          props.begin(),
+                          props.end(),
+                          0U,
+                          [](std::size_t total, property_variant const& pv) {
+                              return total + v5::num_of_const_buffer_sequence(pv);
+                          }
+                      )
+                  )
+                  : 0
+              )
+          )
+    {
+        add_packet_id_to_buf<PacketIdBytes>::apply(packet_id_, packet_id);
+        auto pb = variable_bytes(property_length_);
+        for (auto e : pb) {
+            property_length_buf_.push_back(e);
+        }
+
+        remaining_length_ =
+            PacketIdBytes +       // packet id
+            (
+                reason_code ?
+                (
+                    1 +                   // reason code
+                    property_length_buf_.size() +
+                    property_length_
+                )
+                : 0
+            );
+
+        auto rb = remaining_bytes(remaining_length_);
+        for (auto e : rb) {
+            remaining_length_buf_.push_back(e);
+        }
+    }
+
+    /**
+     * @brief Create const buffer sequence
+     *        it is for boost asio APIs
+     * @return const buffer sequence
+     */
+    std::vector<as::const_buffer> const_buffer_sequence() const {
+        std::vector<as::const_buffer> ret;
+        ret.reserve(num_of_const_buffer_sequence());
+
+        ret.emplace_back(as::buffer(&fixed_header_, 1));
+        ret.emplace_back(as::buffer(remaining_length_buf_.data(), remaining_length_buf_.size()));
+        ret.emplace_back(as::buffer(packet_id_.data(), packet_id_.size()));
+
+        if (reason_code_) {
+            ret.emplace_back(as::buffer(&reason_code_.value(), 1));
+            ret.emplace_back(as::buffer(property_length_buf_.data(), property_length_buf_.size()));
+
+            for (auto const& p : props_) {
+                v5::add_const_buffer_sequence(ret, p);
+            }
+        }
+        return ret;
+    }
+
+    /**
+     * @brief Get whole size of sequence
+     * @return whole size
+     */
+    std::size_t size() const {
+        return 1 + remaining_length_buf_.size() + remaining_length_;
```

Comment:
Should probably document where the 1 comes from.

Reply:
Yes. I will add comments for all magic numbers.

Reply:
Fixed.